File: paulis.py
#@+leo-ver=4-thin
#@+node:cog.20080522002115.2:@thin paulis.py
from numpy import array, complex128

__all__ = ["I","X","Y","Z"]

Z = array([1,0,0,-1],complex128).reshape(2,2)
Y = array([0,1j,-1j, 0],complex128).reshape(2,2)
X = array([0,1,1, 0],complex128).reshape(2,2)
I = array([1,0,0, 1],complex128).reshape(2,2)
#@-node:cog.20080522002115.2:@thin paulis.py
#@-leo

File: sources/VMPS/Qubits/Paulis.hs
-- @+leo-ver=4-thin
-- @+node:gcross.20091118213523.1832:@thin Paulis.hs
-- @@language Haskell

module VMPS.Qubits.Paulis (pI,pX,pY,pZ) where

import Data.Complex
import Data.Vec

import VMPS.Qubits.Operators

i :: Complex Double
i = (0 :+ 1)

pI :: SingleQubitOperator
pX :: SingleQubitOperator
pY :: SingleQubitOperator
pZ :: SingleQubitOperator

pI = SQO (  1 :.  0
         :. 0 :.  1
   :. () )

pX = SQO (  0 :.  1
         :. 1 :.  0
   :. () )

pY = SQO (  0 :.(-i)
         :. i :.  0
   :. () )

pZ = SQO (  1 :.  0
         :. 0 :.(-1)
   :. () )
-- @-node:gcross.20091118213523.1832:@thin Paulis.hs
-- @-leo
